rust/src/write.rs

```diff
@@ -389,6 +440,10 @@ impl<'a, W: Write + Seek> Writer<'a, W> {
         // (That would leave it in an unspecified state if we bailed here!)
         // Instead briefly swap it out for a null writer while we set up the chunker
         // The writer will only be None if finish() was called.
+        if !self.options.use_chunks {
+            unreachable!("Trying to write to a chunk when chunking is disabled")
```

Comment:
I would use panic!, not unreachable!. In English the word communicates your meaning more clearly, but to the compiler it means something else.

In fact this could be an `assert!(self.options.use_chunks, "Trying to write to a chunk when chunking is disabled");`

Reply:
Done. [unreachable!](https://doc.rust-lang.org/std/macro.unreachable.html#panics) is documented to be syntactic sugar around `panic!`, but I'm happy to reserve it's use for cases where it's locally obvious why a property is always true.

Reply:
You're right, I was thinking of https://doc.rust-lang.org/std/hint/fn.unreachable_unchecked.html#

I typically favor assert for these kinds of checks, but that's just my preference. Thanks for making the change.